lib/rubygems/specification.rb

```diff
@@ -1228,11 +1232,7 @@ def self.required_attributes
   def self.reset
     @@dirs = nil
     Gem.pre_reset_hooks.each {|hook| hook.call }
-    @@all = nil
-    @@stubs = nil
-    @@stubs_by_name = {}
-    @@spec_with_requirable_file = {}
-    @@active_stub_with_requirable_file = {}
+    _clear_specs
```

Comment:
No, only reason was to be consistent with the method `_clear_load_cache` method called right below. But I agree that using proper ruby keywords is better. Only thing, I went with `private_class_method` since that's what seems to be used everywhere else.